### DISCUSSION TITLE
Auto translate by gtranslate in nuxt2

### DISCUSSION THREAD

**Discussion by ReSword** [nuxt/nuxt#22957] [Questions (Nuxt 2)]
hi,
I have a web page built based on nuxt2 and I now want it to be able to automatically translate the content on the page, our team chose [gtranslate](https://cn.gtranslate.io/)

we found that Google returns the translated html, but the content on the page is re-rendered with the original text

can this behavior be avoided?

this is our website link  [http://metabigbuyer.com/](http://metabigbuyer.com/) and  [http://ar.metabigbuyer.com/](http://ar.metabigbuyer.com/) translated by gtranslate

**Comment by Im-Lzx** [comment]
> hi, I have a web page built based on nuxt2 and I now want it to be able to automatically translate the content on the page, our team chose [gtranslate](https://cn.gtranslate.io/)
> 
> we found that Google returns the translated html, but the content on the page is re-rendered with the original text
> 
> can this behavior be avoided?
> 
> this is our website link http://metabigbuyer.com/ and http://ar.metabigbuyer.com/ translated by gtranslate

hi,
I have the same problem. How did you solve it? Can you give me some help?

**Reply by ReSword** [reply]
I use the thirdpaty translate api to resolve that, and  translate text in `fetch` funtion, you can also use store to optimization it and reduce ajax request, or in the back-end cache the translate result

**Reply by bohecola** [reply]
My [solution](https://github.com/nuxt/nuxt/discussions/22957#discussioncomment-8819121) is in the comments below, hope it helps you.

**Comment by bohecola** [comment]
I encountered the same issue, but through my debugging, I found that it could be resolved by modifying the Vue source code. 

This issue consists of two parts. One part involves the text nodes rendered by `<div>{{ text }}</div>` or `<div> this is a text </div>`, and the other part involves serialized HTML rendered through `<div v-html="richTextHtml"></div>`.

In my project, I achieved this by commenting out three lines of code in the `node_modules/vue/dist/vue.runtime.esm.js` file, and I have posted the image below.

![WX20240317-224247@2x](https://github.com/nuxt/nuxt/assets/68556445/5c58d0fd-d14a-439f-8b7b-e0a64e68f9ce)

Through the source code, I found that the issue seems to be caused by a mismatch between the real DOM nodes translated by GTranslate and the content of the virtual DOM nodes. This results in the content within the DOM nodes being replaced back with the data from the original virtual DOM, which has not been translated by GTranslate.

Moreover, you can use the [patch-package](https://www.npmjs.com/package/patch-package) npm module to apply modifications to the source code in node_modules, ensuring that these modifications are applied to node_modules each time you run `npm i`.

For example, after you have modified the `node_modules/vue/dist/vue.runtime.esm.js` file, run `npx patch-package vue`, and then add the following script in your `package.json`:

<div class="highlight highlight-source-diff"><pre tabindex="0"> "scripts": {
<span class="pl-mi1"><span class="pl-mi1">+</span>  "postinstall": "patch-package"</span>
 }</pre></div>

I hope my solution is helpful to you.

**Reply by ReSword** [reply]
Thank your idea, I'll try it.

**Reply by jieceng** [reply]
Hello, I have also encountered this issue, but will this change affect the original functionality

**Reply by bohecola** [reply]
it will lose reactivity after the mounted hook.

**Comment by webYangFeng** [comment]
How is your /zh subdirectory implementedI followed the gtranslate official documentation to implement the subdirectory and got a 404 error
